service/cachemdw/cache.go

```diff
@@ -0,0 +1,171 @@
+package cachemdw
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"time"
+
+	"github.com/kava-labs/kava-proxy-service/clients/cache"
+	"github.com/kava-labs/kava-proxy-service/decode"
+	"github.com/kava-labs/kava-proxy-service/logging"
+)
+
+type EVMClient interface {
```

Comment:
1. missing comments for public functions and struct throughout
2. we shouldn't have multiple EVMClients in this repo, one of them is certainly misnamed

Reply:
@galxy25 fixed